src/App.js

```diff
@@ -19,6 +19,26 @@ class App extends Component {
     };
   }
 
+  removeItem = (id) => {
+    const { shoppingCartItens } = this.state;
+    const index = shoppingCartItens.findIndex((product) => product.id === id);
+    shoppingCartItens.splice(index, 1);
+    this.setState({ shoppingCartItens });
+  }
+
+  changeItemQuantity = (id, bool) => {
+    const { shoppingCartItens } = this.state;
+    const index = shoppingCartItens.findIndex((product) => product.id === id);
+    if (bool) {
+      shoppingCartItens[index].quantity += 1;
+    } else if (shoppingCartItens[index].quantity > 1) {
+      shoppingCartItens[index].quantity -= 1;
+    } else {
+      this.removeItem(id);
```

Comment:
Gabriel, o que acha de substituir a chamada da funcao removeItem pelo splice? Vai funcionar do mesmo jeito e ganhamos um pouquinho em performance!

```suggestion
      shoppingCartItens.splice(index, 1);
```